gp-nested-forms/gpnf-auto-attach-child-files.php

```diff
@@ -11,68 +11,134 @@
+			$upload_fields = GFCommon::get_fields_by_type( GFAPI::get_form( $field->gpnfForm ), array( 'fileupload' ) );
+			$child_entries = $parent_entry->get_child_entries( $field->id );
+			$upload_root   = GFFormsModel::get_upload_root();
 
-				$attachment_urls = rgar( $child_entry, $upload_field->id );
-				if ( empty( $attachment_urls ) ) {
-					continue;
-				}
+			foreach ( $child_entries as $child_entry ) {
+				foreach ( $upload_fields as $upload_field ) {
+					if ( ! $this->is_applicable_upload_field( $upload_field ) ) {
+						continue;
+					}
 
-				$attachment_urls = $upload_field->multipleFiles ? json_decode( $attachment_urls, true ) : array( $attachment_urls );
+					$attachment_urls = rgar( $child_entry, $upload_field->id );
+					if ( empty( $attachment_urls ) ) {
+						continue;
+					}
 
-				foreach ( $attachment_urls as $attachment_url ) {
-					$attachment_url = preg_replace( '|^(.*?)/gravity_forms/|', $upload_root, $attachment_url );
-					$attachments[]  = $attachment_url;
+					$attachment_urls = $upload_field->multipleFiles ? json_decode( $attachment_urls, true ) : array( $attachment_urls );
+
+					foreach ( $attachment_urls as $attachment_url ) {
+						$attachment_url = preg_replace( '|^(.*?)/gravity_forms/|', $upload_root, $attachment_url );
+						$attachments[]  = $attachment_url;
+					}
 				}
 			}
 		}
+
+		return $notification;
+	}
+
+	function is_applicabale_notification( $notification ) {
+		return empty( $this->notification_ids ) || in_array( $notification['id'], $this->notification_ids );
+	}
+
+	function is_applicable_upload_field( $upload_field ) {
+		return empty( $this->child_form_upload_field_ids ) || in_array( $upload_field->id, $this->child_form_upload_field_ids );
 	}
+}
+
+/**
+ *  Notes:
+ * - If notifications IDs are specified this snippet will only work for those specific forms.
+ * - If child form upload field IDs are specified, only files from those fields will be attached.
+ * - If the parent form has upload fields, they will be attached instead of the child form's fields.
+ * - If none of these conditions are present, child form uploads will be attached by default.
+ *
+ * Configuration:
+ *
+ * 1) Ensure that the parent form's notification has the "Attach uploaded fields to notification" checked.
+ * 2) [Optional] Modify one ore more of the following examples to limit the snippet to specific notification IDs or child form upload field IDs.
+ *
+ * 💡 Notification IDs can be retrieved from the nid parameter in the URL when editing a notification.
+*/
+
+# -------------------------------------------------
+# Examples
+# -------------------------------------------------
+
+# Attach all child form uploads to all notifications.
+// new GPNF_Auto_Attach_Child_Files();
+
+# Attach all uploads from child forms to *only* the notification with id "63331f3fcf7f0".
+// new GPNF_Auto_Attach_Child_Files(
+// 	array(
+// 		'notification_ids' => array( '63331f3fcf7f0' ),
+// 	)
+// );
```

Comment:
Just for the sake of consistency with our other snippets, can you put the `array` on the same line as the class name?

e.g.

```suggestion
# Attach all uploads from child forms to *only* the notification with id "63331f3fcf7f0".
// new GPNF_Auto_Attach_Child_Files( array(
//    'notification_ids' => array( '63331f3fcf7f0' ),
// ) );
```

Reply:
Yup! Done